src/transformers/pipelines/table_question_answering.py

```diff
@@ -83,25 +96,25 @@ def __init__(self, args_parser=TableQuestionAnsweringArgumentHandler(), *args, *
         super().__init__(*args, **kwargs)
         self._args_parser = args_parser
 
-        if self.framework == "tf":
-            raise ValueError("The TableQuestionAnsweringPipeline is only available in PyTorch.")
-
-        self.check_model_type(MODEL_FOR_TABLE_QUESTION_ANSWERING_MAPPING)
+        self.check_model_type(
+            TF_MODEL_FOR_TABLE_QUESTION_ANSWERING_MAPPING
+            if self.framework == "tf"
+            else MODEL_FOR_TABLE_QUESTION_ANSWERING_MAPPING
+        )
 
         self.aggregate = bool(getattr(self.model.config, "aggregation_labels")) and bool(
             getattr(self.model.config, "num_aggregation_labels")
         )
 
     def batch_inference(self, **inputs):
-        with torch.no_grad():
-            return self.model(**inputs)
+        return self.model(**inputs)
 
     def sequential_inference(self, **inputs):
         """
         Inference used for models that need to process sequences in a sequential fashion, like the SQA models which
         handle conversational query related to a table.
         """
-        with torch.no_grad():
```

Comment:
It looks like you removed the `torch.no_grad()` context here - this doesn't affect the TF code, but will this cause issues when this code is used with Torch?

Reply:
No. 
https://github.com/huggingface/transformers/blob/a2dec768a27ab7520d4ae4f5f72643043f305fd3/src/transformers/pipelines/base.py#L863-L875

pipeline_base_forward -> table_qa_forward -> sequential_inference. 
Torch code will run with `torch.no_grad()`